web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabase.test.tsx

```diff
@@ -102,7 +102,8 @@ describe('test EnrollRdsDatabase.tsx', () => {
     expect(ctx.databaseService.fetchDatabases).not.toHaveBeenCalled();
   });
 
-  test('with rds database result, makes a fetch request for db servers', async () => {
+  // TODO: Fix flaky network error failure in this test
+  test.skip('with rds database result, makes a fetch request for db servers', async () => {
```

Comment:
It was `userEventService.catpureDiscoverEvent` that was causing the error. I also refactored the test so that it mocks things before attempting to render the component.

<details>

```diff
diff --git a/web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabase.test.tsx b/web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabase.test.tsx
index d87688d5f6..a113180a99 100644
--- a/web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabase.test.tsx
+++ b/web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabase.test.tsx
@@ -36,59 +36,27 @@ import {
   DatabaseLocation,
 } from 'teleport/Discover/SelectResource';
 import { FeaturesContextProvider } from 'teleport/FeaturesContext';
+import { userEventService } from 'teleport/services/userEvent';
 
 import { EnrollRdsDatabase } from './EnrollRdsDatabase';
 
 describe('test EnrollRdsDatabase.tsx', () => {
-  const ctx = createTeleportContext();
-  const discoverCtx: DiscoverContextState = {
-    agentMeta: {
-      integration: {
-        kind: 'aws-oidc',
-        name: 'aws-oidc-integration',
-        resourceType: 'integration',
-        spec: {
-          roleArn: 'arn-123',
-        },
-        statusCode: IntegrationStatusCode.Running,
-      },
-    } as any,
-    currentStep: 0,
-    nextStep: jest.fn(x => x),
-    prevStep: () => null,
-    onSelectResource: () => null,
-    resourceSpec: {
-      dbMeta: {
-        location: DatabaseLocation.Aws,
-        engine: DatabaseEngine.AuroraMysql,
-      },
-    } as any,
-    viewConfig: null,
-    exitFlow: null,
-    indexedViews: [],
-    setResourceSpec: () => null,
-    updateAgentMeta: jest.fn(x => x),
-    emitErrorEvent: () => null,
-    emitEvent: () => null,
-    eventState: null,
-  };
-
   beforeEach(() => {
-    jest
-      .spyOn(ctx.databaseService, 'fetchDatabases')
-      .mockResolvedValue({ agents: [] });
-  });
-
-  afterEach(() => {
-    jest.clearAllMocks();
+    jest.restoreAllMocks();
   });
 
   test('without rds database result, does not attempt to fetch db servers', async () => {
-    renderRdsDatabase(ctx, discoverCtx);
+    const { ctx, discoverCtx } = getMockedContexts();
     jest
       .spyOn(integrationService, 'fetchAwsRdsDatabases')
       .mockResolvedValue({ databases: [] });
 
+    render(
+      <Wrapper ctx={ctx} discoverCtx={discoverCtx}>
+        <EnrollRdsDatabase />
+      </Wrapper>
+    );
+
     // select a region from selector.
     const selectEl = screen.getByLabelText(/aws region/i);
     fireEvent.focus(selectEl);
@@ -102,13 +70,18 @@ describe('test EnrollRdsDatabase.tsx', () => {
     expect(ctx.databaseService.fetchDatabases).not.toHaveBeenCalled();
   });
 
-  // TODO: Fix flaky network error failure in this test
-  test.skip('with rds database result, makes a fetch request for db servers', async () => {
-    renderRdsDatabase(ctx, discoverCtx);
+  test('with rds database result, makes a fetch request for db servers', async () => {
+    const { ctx, discoverCtx } = getMockedContexts();
     jest.spyOn(integrationService, 'fetchAwsRdsDatabases').mockResolvedValue({
       databases: mockAwsDbs,
     });
 
+    render(
+      <Wrapper ctx={ctx} discoverCtx={discoverCtx}>
+        <EnrollRdsDatabase />
+      </Wrapper>
+    );
+
     // select a region from selector.
     const selectEl = screen.getByLabelText(/aws region/i);
     fireEvent.focus(selectEl);
@@ -123,20 +96,65 @@ describe('test EnrollRdsDatabase.tsx', () => {
   });
 });
 
-function renderRdsDatabase(
-  ctx: TeleportContext,
-  discoverCtx: DiscoverContextState
+function getMockedContexts() {
+  const ctx = createTeleportContext();
+  const discoverCtx: DiscoverContextState = {
+    agentMeta: {
+      integration: {
+        kind: 'aws-oidc',
+        name: 'aws-oidc-integration',
+        resourceType: 'integration',
+        spec: {
+          roleArn: 'arn-123',
+        },
+        statusCode: IntegrationStatusCode.Running,
+      },
+    } as any,
+    currentStep: 0,
+    nextStep: jest.fn(x => x),
+    prevStep: () => null,
+    onSelectResource: () => null,
+    resourceSpec: {
+      dbMeta: {
+        location: DatabaseLocation.Aws,
+        engine: DatabaseEngine.AuroraMysql,
+      },
+    } as any,
+    viewConfig: null,
+    exitFlow: null,
+    indexedViews: [],
+    setResourceSpec: () => null,
+    updateAgentMeta: jest.fn(x => x),
+    emitErrorEvent: () => null,
+    emitEvent: () => null,
+    eventState: null,
+  };
+  jest
+    .spyOn(ctx.databaseService, 'fetchDatabases')
+    .mockResolvedValue({ agents: [] });
+  jest
+    .spyOn(userEventService, 'captureDiscoverEvent')
+    .mockResolvedValue(undefined as never);
+
+  return { ctx, discoverCtx };
+}
+
+function Wrapper(
+  props: React.PropsWithChildren<{
+    ctx: TeleportContext;
+    discoverCtx: DiscoverContextState;
+  }>
 ) {
-  return render(
+  return (
     <MemoryRouter
       initialEntries={[
         { pathname: cfg.routes.discover, state: { entity: 'database' } },
       ]}
     >
-      <ContextProvider ctx={ctx}>
+      <ContextProvider ctx={props.ctx}>
         <FeaturesContextProvider value={[]}>
-          <DiscoverProvider mockCtx={discoverCtx}>
-            <EnrollRdsDatabase />
+          <DiscoverProvider mockCtx={props.discoverCtx}>
+            {props.children}
           </DiscoverProvider>
         </FeaturesContextProvider>
       </ContextProvider>
```

</details>